% NOTE -- ONLY EDIT THE .Rnw FILE!!!  The .tex file is
% likely to be overwritten.
% \VignetteIndexEntry{SNPchip Vignette}
% \VignetteKeywords{copy number, genotype, SNP}
%\VignettePackage{SNPchip}
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\documentclass[]{article}

\usepackage{amsmath,fullpage}
\usepackage{natbib}
\usepackage{hyperref}
\usepackage{setspace}
\parindent 0in  % Left justify
\newcommand{\scscst}{\scriptscriptstyle}
\newcommand{\scst}{\scriptstyle}
\usepackage{natbib}

\newcommand{\Rpackage}[1]{\textit{#1}}
\newcommand{\Rfunction}[1]{\texttt{#1}}
\newcommand{\Robject}[1]{\texttt{#1}}

\textwidth=6.2in
\textheight=8.5in
%\parskip=.3cm
\oddsidemargin=.1in
\evensidemargin=.1in
\headheight=-.3in

\title{Tools for high throughput SNP chip data}
\author{Robert Scharpf}

\begin{document}

<<options, echo=FALSE, hide=TRUE>>=
options(width=60)
@

\maketitle

\section*{Introduction}

\Rpackage{SNPchip} includes methods for visualizing copy number and
inferred copy number alterations from high-throughput genotyping
arrays.

\section{Simple Usage}

<<package, hide=TRUE>>=
library(SNPchip)
data(oligoSet)
@

<<subsetting>>=
snpset <- oligoSet[chromosome(oligoSet) %in% as.character(1:3), 1:4]
@

Plot the first few chromosomes for samples 1-4:

%<<plot1, fig=TRUE, width=9, height=6>>=
%##print(graph.par)
%@

The samples are plotted by row. For each sample, the copy number
(vertical axis) is plotted against the physical position of the SNP in
the chromosome.  Here, the chromosome labels are plotted beneath the
cytobands.

\section{Examples}

\subsection{Genome-wide plots for multiple samples}
A genome-wide view of copy number and genotype calls versus physical
position can be made using \Rfunction{plot}.  Here, we plot
autosomes 1-22 of samples 1 - 3 in the object
\Robject{oligoSet}:

Note that we suppress the cytobands in the above plot (the resolution
is too poor at this level) by the argument \Robject{add.cytoband} in
the function \Rfunction{getPar}.  The default plot layout generally
works well, but can be adjusted through additional arguments to par
and layout.


A plot of just the p-arm in sample 2 of chromosome 1.

%<<plotP, fig=TRUE, width=8, height=7>>=
%##print(graph.par)
%@

Note that the cytoband is automatically subsetted appropriately. Had
we instead specified \texttt{use.chromosome.size=TRUE}, the x-axis
limits would include the entire chromosome (and cytoband) though only
the SNPs on the p-arm would be plotted.

Adding a legend for the genotypes
%<<addLegend, fig=TRUE, width=8, height=5>>=
<<addLegend>>=
data(oligoSet)
x <- oligoSet[chromosome(oligoSet) == "1", 1]
@

%Alternatively, plot the cytoband on top and then add the legend to the plot by hand.

\subsection{Plotting cytoband}

To plot the cytoband of chromosome 1,

<<cytoband, fig=TRUE, width=10, height=2>>=
plotCytoband("1")
@

%\subsection{Smoothing example}

Hidden Markov models for objects of class \Robject{SnpSet},
\Robject{CopyNumberSet}, and \Robject{oligoSnpSet} are available in the
package \Rpackage{VanillaICE} available at Bioconductor.  A more
detailed description of the hidden Markov models fit in the
\Rpackage{VanillaICE} package are discussed elsewhere
\citep{Scharpf2007a}.  Here we provide an example of a lowess smoother
for copy number estimates.  The following code chunk first assigns
heterozygous calls to the integer 1 and homozygous calls to the integer
zero.  It follows that regions of deletions will have homozygous calls
of zero. We simulated a deletion of 50 consecutive SNPs and converted
the \Robject{oligoSet} to a list where each element in the list is
an \Robject{oligoSnpSet} object for one chromosome.


%Alternatively, one may obtain the NetAffx annotation saved as an R
%object here:
%<<netAffxAnnotation, eval=FALSE, echo=FALSE>>=
%path <- "http://biostat.jhsph.edu/~iruczins/publications/sm/2006.scharpf.bioinfo"
%try(load(url(paste(path, "/mapping/mapping10k.rda", sep=""))))
%colnames(mapping10k$annotation)
%@

\section{Session Information}
The version number of R and packages loaded for generating the vignette were:

<<echo=FALSE, results=tex>>=
toLatex(sessionInfo())
@

%\bibliography{refs}
%\bibliographystyle{plain}

\end{document}
